### PR TITLE
Add `log1p` in `CommonNeighbors` to solve loss spikes

### PR DESCRIPTION
**Pull request by tizianocitro**
The `CommonNeighbors` model produces unstable loss values, especially with `Aggregation.SUM`. The root cause was a magnitude mismatch between the scores and what `BCEWithLogitsLoss` expects.

The problem is `CommonNeighborsScorer` returning raw common-neighbor counts (non-negative, unbounded integers). These are passed directly to `BCEWithLogitsLoss`. `MEAN` and `MIN` produce smaller scores so the effect is less severe, but `SUM` produces very high counts that fully saturate the sigmoid.

The fix is adding `log1p` in the model:

```python
scores = self.scorer.score_batch(hyperedge_index, node_to_neighbors)
torch.log1p(scores, out=scores)
return scores
```

### CODE REVIEW

**Comment by tizianocitro** [comment]
`log1p(x) = log(1 + x)` compresses the score range into a logit-friendly scale:

| Raw score | After log1p | sigmoid |
|-----------|-------------|---------|
| 0         | 0.0         | 0.50    |
| 1         | 0.69        | 0.67    |
| 10        | 2.40        | 0.91    |
| 100       | 4.62        | 0.99    |
| 1000      | 6.91        | 0.999   |

This is might be the right approach because:
- **Monotonic**: preserves score ranking, so rank-based metrics (AUC, AP) are unaffected.
- **Minimal change**: one line, applied at the consumption site, not in the scorer.
